golem-component-service-base/tests/services_tests.rs

```diff
@@ -0,0 +1,453 @@
+#[cfg(test)]
+mod tests {
+    use golem_service_base::auth::DefaultNamespace;
+    use golem_service_base::config::{ComponentStoreLocalConfig, DbPostgresConfig, DbSqliteConfig};
+    use golem_service_base::db;
+
+    use golem_common::model::ComponentId;
+    use golem_component_service_base::model::Component;
+    use golem_component_service_base::repo::component::{ComponentRepo, DbComponentRepo};
+    use golem_component_service_base::service::component::{
+        create_new_component, ComponentService, ComponentServiceDefault,
+    };
+    use golem_component_service_base::service::component_compilation::{
+        ComponentCompilationService, ComponentCompilationServiceDisabled,
+    };
+    use golem_service_base::model::ComponentName;
+    use golem_service_base::service::component_object_store;
+    use std::sync::Arc;
+    use testcontainers::clients::Cli;
+    use testcontainers::{Container, RunnableImage};
+    use testcontainers_modules::postgres::Postgres;
+    use uuid::Uuid;
+
+    fn start_docker_postgres<'d>(docker: &'d Cli) -> (DbPostgresConfig, Container<'d, Postgres>) {
+        let image = RunnableImage::from(Postgres::default()).with_tag("14.7-alpine");
+        let container: Container<'d, Postgres> = docker.run(image);
+
+        let config = DbPostgresConfig {
+            host: "localhost".to_string(),
+            port: container.get_host_port_ipv4(5432),
+            database: "postgres".to_string(),
+            username: "postgres".to_string(),
+            password: "postgres".to_string(),
+            schema: Some("test".to_string()),
+            max_connections: 10,
+        };
+
+        (config, container)
+    }
+
+    struct SqliteDb {
+        db_path: String,
+    }
+
+    impl Default for SqliteDb {
+        fn default() -> Self {
+            Self {
+                db_path: format!("/tmp/golem-component-{}.db", uuid::Uuid::new_v4()),
+            }
+        }
+    }
+
+    impl Drop for SqliteDb {
+        fn drop(&mut self) {
+            std::fs::remove_file(&self.db_path).unwrap();
+        }
+    }
+
+    #[tokio::test]
+    pub async fn test_with_postgres_db() {
+        let cli = Cli::default();
+        let (db_config, _container) = start_docker_postgres(&cli);
+
+        db::postgres_migrate(&db_config, "tests/db/migration/postgres")
```

Comment:
You can specify path to real migrations here as `../golem-component-service/...`.